api/jsonapi.go

```diff
@@ -533,20 +533,100 @@ func (i *jsonAPIHandler) PUTImage(w http.ResponseWriter, r *http.Request) {
 func (i *jsonAPIHandler) POSTListing(w http.ResponseWriter, r *http.Request) {
```

Comment:
What about something like this for error handling? Just to make things a bit cleaner IMO.

```go
func (i *jsonAPIHandler) POSTListing(w http.ResponseWriter, r *http.Request) {
  	w.Header().Add("Content-Type", "application/json")

    var err error
    defer func(){
      if err != nil {
        w.WriteHeader(http.StatusBadRequest)
        fmt.Fprintf(w, `{"success": false, "reason": "%s"}`, err)
      }
    }()

    //...

    err = i.node.SetListingInventory(ld.Listing, ld.Inventory)
    if err != nil {
    	return
    }

    f, err := os.Create(path.Join(listingPath, "listing.json"))
    if err != nil {
      return
    }

    //...
}
```